src/App.tsx

```diff
@@ -15,36 +17,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export enum SortType {
+  SORT_BY_LENGTH = 'SORT_BY_LENGTH',
+  SORT_ALPHABETICALLY = 'SORT_ALPHABETICALLY',
+  DEFAULT = 'DEFAULT',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  reverse: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SortType.SORT_BY_LENGTH:
+          return a.length - b.length;
+
+        case SortType.SORT_ALPHABETICALLY:
+          return a.localeCompare(b);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.DEFAULT);
+  const [reverse, setReverse] = useState(false);
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
Boolean variables should start from `is` or `has`